File: Java/Algorithms/Math/Factorial.java
public class Factorial {
    public static void main(String[] args) {
        System.out.println(Fact(5));
    }

    static int Fact(int n){
        if(n==0)
            return 1;
        else{
            return(n*Fact(n-1));
        }
        
    }
}

File: Go/factorial.go
// This program outputs the factorial of a number.
package main
import "fmt"


var fact uint64 = 1 

var i int = 1
var n int


func factorial(n int) uint64 {   
    if(n < 0){
        fmt.Print("Factorial of negative number doesn't exist.")    
    }else{        
        for i:=1; i<=n; i++ {
            fact *= uint64(i)  
        }

    }    
    return fact  
}

func main(){    
    fmt.Print("Enter a positive integer: ")
    fmt.Scan(&n)   
    fmt.Print("Factorial is: ",factorial(n))
}
